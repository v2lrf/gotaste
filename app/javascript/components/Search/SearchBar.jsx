import React from 'react'

const SearchBar = () => (
  <div className="bg-red-lightest py-12 px-4">
    <div className="container mx-auto px-4">
      <div className="w-1/2 font-serif">
        <h1 className="text-5xl font-black-darker font-black">
          Find den bedste vin
        </h1>
      </div>

      <div className="flex py-4 mr-6">
        <input
          className="w-2/3 py-6 px-4 mr-2 border border-grey-light rounded text-grey-darkest text-lg"
          type="text"
          placeholder="F.eks. Vesterbro, Nørrebro eller bare København..."
        />
        <button
          type="button"
          className="w-1/3 font-sans py-4 px-4 border-transparent bg-red-light text-white rounded uppercase font-bold text-sm"
        >
          Søg
        </button>
      </div>
    </div>
  </div>
)

export default SearchBar
