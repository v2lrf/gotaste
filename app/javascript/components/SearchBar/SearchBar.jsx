import React, { Component } from 'react'
import PropTypes from 'prop-types'
import Places from 'places.js'

class SearchBar extends Component {
  componentDidMount() {
    const { onChange } = this.props
    const placesAutocomplete = Places({
      appId: 'plLOLX7HJZL4',
      apiKey: '8d3d74f780371b26a3f07b788a85e592',
      container: this.autoCompletePlace
    })
    placesAutocomplete.on('change', event => onChange(event))
  }

  render() {
    return (
      <div>
        <input
          type="text"
          ref={input => {
            this.autoCompletePlace = input
          }}
        />
      </div>
    )
  }
}

SearchBar.propTypes = {
  onChange: PropTypes.func.isRequired
}

export default SearchBar
