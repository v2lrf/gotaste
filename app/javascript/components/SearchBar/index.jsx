import React, { Component } from 'react'
import PropTypes from 'prop-types'
import Places from 'places.js'

export class SearchBar extends Component {
  componentDidMount() {
    const { onChange } = this.props
    const placesAutocomplete = Places({
      appId: process.env.ALGOLIA_PLACES_APP_ID,
      apiKey: process.env.ALGOLIA_PLACES_API_KEY,
      container: this.autoCompletePlace,
      countries: ['DK']
    })
    placesAutocomplete.on('change', event => onChange(event))
  }

  render() {
    return (
      <div>
        <input
          type="text"
          ref={input => {
            this.autoCompletePlace = input
          }}
        />
      </div>
    )
  }
}

SearchBar.propTypes = {
  onChange: PropTypes.func.isRequired
}

export default SearchBar
