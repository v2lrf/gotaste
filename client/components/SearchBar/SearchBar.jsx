import React, { Component } from 'react'
import PropTypes from 'prop-types'
import Places from 'places.js'

class SearchBar extends Component {
  componentDidMount() {
    const { onChange } = this.props

    const placesAutocomplete = Places({
      appId: process.env.ALGOLIA_PLACES_APP_ID,
      apiKey: process.env.ALGOLIA_PLACES_API_KEY,
      container: this.autoCompletePlace,
      countries: ['DK']
    })

    placesAutocomplete.on('change', event => {
      const {
        suggestion: { latlng }
      } = event
      onChange({
        latitude: latlng.lat,
        longitude: latlng.lng
      })
    })
  }

  render() {
    return (
      <div>
        <input
          type="text"
          ref={input => {
            this.autoCompletePlace = input
          }}
          placeholder="Indtast addresse"
        />
      </div>
    )
  }
}

SearchBar.propTypes = {
  onChange: PropTypes.func.isRequired
}

export default SearchBar
