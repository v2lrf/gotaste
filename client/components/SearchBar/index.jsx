/**
 * Please note that eslint is currently disabled for this file,
 * because dynamic imports is still officially an experimental
 * feature.
 * ESlint are not implementing rules for features until they
 * are officially implemented.
 * See: https://github.com/eslint/eslint#what-about-experimental-features
 */
import dynamic from 'next/dynamic'
import { GhostCard } from '../Ghost'

const SearchBar = dynamic({
  loader: () => import('./SearchBar'),
  loading: () => <GhostCard height={50} />,
  ssr: false
})

export default SearchBar
