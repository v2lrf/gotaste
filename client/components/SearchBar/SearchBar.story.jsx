import React from 'react'
import { storiesOf } from '@storybook/react'

import SearchBar from '.'

storiesOf('SearchBar', module).add('Default', () => (
  <div className="p-16 bg-grey-lighter">
    {/* eslint-disable-next-line no-console */}
    <SearchBar onChange={event => console.log(event)} />
  </div>
))
