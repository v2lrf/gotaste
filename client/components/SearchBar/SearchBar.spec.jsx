import React from 'react'
import SearchBar from './SearchBar'

describe('SearchBar', () => {
  let component
  let props
  beforeEach(() => {
    props = {
      onChange: jest.fn()
    }

    component = render(<SearchBar {...props} />)
  })

  describe('Snapshots', () => {
    it('should match the snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })
})
