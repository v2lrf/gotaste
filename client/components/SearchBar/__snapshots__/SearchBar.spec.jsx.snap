// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar Snapshots should match the snapshot 1`] = `
<div>
  <input
    placeholder="Indtast addresse"
    type="text"
  />
</div>
`;
